simulation/src/test/java/com/palantir/dialogue/core/SimulationTest.java

```diff
@@ -367,6 +370,43 @@ public void live_reloading() {
                 .run();
     }
 
+    @Test
+    public void uncommon_flakes() {
+        servers = servers(
+                SimulationServer.builder()
+                        .serverName("fast0")
+                        .simulation(simulation)
+                        .handler(h -> h.response(respond500Every(100)).responseTime(Duration.ofNanos(1000)))
+                        .build(),
+                SimulationServer.builder()
+                        .serverName("fast1")
+                        .simulation(simulation)
+                        .handler(h -> h.response(respond500Every(100)).responseTime(Duration.ofNanos(1000)))
+                        .build());
+
+        result = Benchmark.builder()
+                .requestsPerSecond(1000)
+                .sendUntil(Duration.ofSeconds(10))
+                .clients(10, i -> strategy.getChannel(simulation, servers))
+                .simulation(simulation)
+                .abortAfter(Duration.ofSeconds(10))
+                .run();
+    }
+
+    private Function<SimulationServer, Response> respond500Every(int reqs) {
+        return new Function<SimulationServer, Response>() {
+            private final AtomicInteger count = new AtomicInteger();
```

Comment:
I've been using an int array with a single element to emphasise this entire thing is single threaded